src/lib/proof-system/zkprogram.ts

```diff
@@ -132,6 +132,21 @@ async function verify(
   );
 }
 
+/**
+ * Asserts that the hash computed from a verification key's data matches the provided hash.
+ * This is used to ensure that the verification key data and hash are consistent.
+ *
+ * @param verificationKey - The verification key whose data and hash to check for consistency
+ * @returns void, throws an error if the computed hash doesn't match the provided hash
+ */
+async function validateVkHash(verificationKey: VerificationKey) {
```

Comment:
This is redundant, this logic already exists in the zkProgram. Instead, you should just constrain `hash` somehow in relation to your state.